federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -724,18 +724,24 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	if !wantsDNSRecords(service.appliedState) {
 		return nil
 	}
-	ensuredCount := 0
+
+	var err error
+	missedCount := 0
+	unensuredCount := 0
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
-				ensuredCount += 1
+				if err = s.ensureDnsRecords(clusterName, service); err != nil {
+					unensuredCount += 1
+				}
+			} else {
+				missedCount += 1
 			}
 		}
 	}
-	if ensuredCount < len(clusterNames) {
-		return fmt.Errorf("Failed to update DNS records for %d of %d clusters for service %v due to missing clients for those clusters",
-			len(clusterNames)-ensuredCount, len(clusterNames), service)
+	if missedCount > 0 || unensuredCount > 0 {
+		return fmt.Errorf("Failed to update DNS records for %d clusters for service %v due to missing clients[missed count: %d] or ensuring DNS records error[unensured count: %d] %v",
+			len(clusterNames), service, missedCount, unensuredCount, err)
```

Comment:
The error of ensureDnsRecords for every cluster maybe different, but here is only the last one. So I'd better not include err in the lockedUpdateDNSRecords func return info. Just log the err in the loop?